core/src/test/java/org/elasticsearch/cluster/action/shard/ShardStateActionTests.java

```diff
@@ -284,6 +285,13 @@ public void onShardFailedFailure(Exception e) {
         assertTrue(failure.get());
     }
 
+    public void testMasterChannelExceptions() {
+        assertTrue(ShardStateAction.isMasterChannelException(new TransportException(new NotMasterException("simulated"))));
+        assertTrue(ShardStateAction.isMasterChannelException(new TransportException(new Discovery.FailedToCommitClusterStateException("simulated"))));
```

Comment:
maybe wrap in a RemoteTransportException? will communicate better why this is here...